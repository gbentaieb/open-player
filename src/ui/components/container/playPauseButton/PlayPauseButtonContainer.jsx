import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PlayPauseButton from '../../presentational/playpausebutton/PlayPauseButton';

import { requestPlay } from '../../../actions/CoreActions';

import { isPlaying } from '../../../utils/playerStates';

function mapStateToProps(state) {
  return {
    isPlaying: isPlaying(state.core.playerState),
  };
}

class PlayPauseButtonContainer extends Component {
  static propTypes = {
    isPlaying: PropTypes.bool.isRequired,
    requestPlay: PropTypes.func.isRequired,
  }

  static defaultProps = {
    /* Add default props */
  }

  playPause = () => {
    this.props.requestPlay(!this.props.isPlaying);
  }

  render() {
    return (
      <PlayPauseButton
        onClick={this.playPause}
        isPlaying={this.props.isPlaying}
      />
    );
  }
}

export default connect(
  mapStateToProps,
  {
    requestPlay,
  },
)(PlayPauseButtonContainer);
