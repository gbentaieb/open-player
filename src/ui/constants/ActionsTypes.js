// Config Actions
export const MERGE_CONFIG = 'MERGE_CONFIG';

// Core Actions
export const SET_PLAYER_STATE = 'SET_PLAYER_STATE';
export const REQUEST_PLAY = 'REQUEST_PLAY';
export const SEEK_TO = 'SEEK_TO';
export const SET_PLAYER_TIMES = 'SET_PLAYER_TIMES';
