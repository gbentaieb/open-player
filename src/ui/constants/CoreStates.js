export const PLAYING = 'PLAYING';
export const SEEKING = 'SEEKING';
export const BUFFERING = 'BUFFERING';
export const STOPPED = 'STOPPED';
export const LOADING = 'LOADING';
export const LOADED = 'LOADED';
export const ENDED = 'ENDED';
export const PAUSED = 'PAUSED';

export const CORE_STATES = [PLAYING, SEEKING, BUFFERING, STOPPED, LOADING, LOADED, ENDED, ENDED];
