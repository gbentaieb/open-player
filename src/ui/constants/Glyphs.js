import playGlyph from '../assets/icons/play.svg';
import pauseGlyph from '../assets/icons/pause.svg';

export const PLAY_GLYPH = playGlyph;
export const PAUSE_GLYPH = pauseGlyph;
