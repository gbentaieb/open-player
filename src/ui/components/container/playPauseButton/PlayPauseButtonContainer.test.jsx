import React from 'react';
import { createMockStore } from 'redux-test-utils';

import PlayPauseButtonContainer from './PlayPauseButtonContainer';
import * as playerStates from '../../../constants/CoreStates';

const defaultStore = {
  core: {
    playerState: playerStates.STOPPED,
  },
};

describe('PlayPauseButtonContainer snapshot', () => {
  test('PlayPauseButtonContainer > Snapshot', () => {
    const store = createMockStore(defaultStore);
    const wrapper = shallow(<PlayPauseButtonContainer store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('PlayPauseButtonContainer methods', () => {
  test('PlayPauseButtonContainer > isPlaying when PAUSED', () => {
    const store = createMockStore(
      {
        ...defaultStore,
        core: { playerState: playerStates.PAUSED },
      },
    );
    const wrapper = shallow(<PlayPauseButtonContainer store={store} />);
    const instance = wrapper.dive().instance();
    expect(instance.props.isPlaying).toEqual(false);
  });

  test('PlayPauseButtonContainer > isPlaying when PLAYING', () => {
    const store = createMockStore(
      {
        ...defaultStore,
        core: { playerState: playerStates.PLAYING },
      },
    );
    const wrapper = shallow(<PlayPauseButtonContainer store={store} />);
    const instance = wrapper.dive().instance();
    expect(instance.props.isPlaying).toEqual(true);
  });
});
