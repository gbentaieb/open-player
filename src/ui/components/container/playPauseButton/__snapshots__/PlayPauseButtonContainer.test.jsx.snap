// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayPauseButtonContainer snapshot PlayPauseButtonContainer > Snapshot 1`] = `
<PlayPauseButtonContainer
  isPlaying={false}
  requestPlay={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": undefined,
    }
  }
/>
`;
